cmd/livepeer/livepeer.go

```diff
@@ -726,7 +728,7 @@ func main() {
 	ec := make(chan error)
 	tc := make(chan struct{})
 	wc := make(chan struct{})
-	msCtx, cancel := context.WithCancel(context.Background())
+	msCtx, cancel := context.WithCancel(ctx)
```

Comment:
Can't we just declare the contexts with cancel once top-level?
Don't we currently redeclare these despite them being the same value?

Reply:
> Can't we just declare the contexts with cancel once top-level?

`msCtx` is used for special handling by the media server and should not be confused with the context it was forked from.

> Don't we currently redeclare these despite them being the same value?

`context.Background()` returns the global context.  It's like using global variables in an application, which we all know is a no-no, especially in cases of concurrency.  Right now it compiles down to the same binary with or without this cleanup, but code readability is a problem.  This patch aims to begin fixing the code readability of the context handling.

context should usually be the same within a function, and may or may not be forked into the functions called from it.  When a function needs specific cancellation handling, that is when we definitely want to fork a given context to a contextWithTimeout, contextWithCancel or contextWithDeadline so as not to affect the context that it was forked from.

For now, this is a step forward.  It begins to improve the readability of context handling, and it has no negative effects.

Reply:
> When a function needs specific cancellation handling, that is when we definitely want to fork a given context to a contextWithTimeout, contextWithCancel or contextWithDeadline so as not to affect the context that it was forked from.

Thanks for clarifying !